lib/rucio/core/did.py

```diff
@@ -43,12 +43,13 @@
 
 if TYPE_CHECKING:
     from collections.abc import Callable, Sequence
-    from typing import Any, Optional, Union
+    from typing import Any, Dict, Optional, Union, List
```

Comment:
```suggestion
    from typing import Any, Optional, Union
```

Built-in `dict` and `list` types should be used instead of `typing.Dict` and `typing.List`